src/connection.rs

```diff
@@ -83,7 +83,7 @@ where
         if connection.is_dead {
             return Ok(None)
         }
-        if connection.streams.len() >= connection.config.max_num_streams {
+        if connection.streams.len() >= connection.config.max_substreams {
```

Comment:
Within this crate the notions used are `Connection` and `Stream`. The sudden use of the term "substream" in configuration is alien to that, so I would prefer to not make this change.